src/Component/Tests/Symfony/EventListener/DeserializeListenerTest.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Resource\Tests\Symfony\EventListener;
+
+use PHPUnit\Framework\TestCase;
+use Prophecy\PhpUnit\ProphecyTrait;
+use Prophecy\Prophecy\ObjectProphecy;
+use Sylius\Resource\Metadata\HttpOperation;
+use Sylius\Resource\Metadata\Operation\HttpOperationInitiatorInterface;
+use Sylius\Resource\Metadata\ResourceMetadata;
+use Sylius\Resource\Symfony\EventListener\DeserializeListener;
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\Serializer\SerializerInterface;
+
+final class DeserializeListenerTest extends TestCase
+{
+    use ProphecyTrait;
+
+    /** @var HttpOperationInitiatorInterface|ObjectProphecy */
+    private $operationInitiator;
```

Comment:
Since we're on php8.0, we should be able to write union types directly in code.
```suggestion
    private HttpOperationInitiatorInterface|ObjectProphecy $operationInitiator;
```